neo4j/auth/cert_test.go

```diff
@@ -0,0 +1,177 @@
+package auth
+
+import (
+	"testing"
+)
+
+var validPassword = "thepassword1"
+var invalidPassword = "invalid_password"
+
+type testCase struct {
+	description string
+	certFile    string
+	keyFile     string
+	password    *string
+}
+
+func TestNewStaticClientCertificateProvider(t *testing.T) {
+	errorTestCases := []testCase{
+		{
+			description: "Invalid cert file path",
+			certFile:    "invalid_path",
+			keyFile:     "./testdata/test_key.pem",
+		},
+		{
+			description: "Invalid key file path",
+			certFile:    "./testdata/test_cert.pem",
+			keyFile:     "invalid_path",
+		},
+		{
+			description: "Incorrect password",
+			certFile:    "./testdata/test_cert.pem",
+			keyFile:     "./testdata/test_key_with_thepassword1.pem",
+			password:    &invalidPassword,
+		},
+		{
+			description: "Missing password",
+			certFile:    "./testdata/test_cert.pem",
+			keyFile:     "./testdata/test_key_with_thepassword1.pem",
+		},
+	}
+
+	t.Run("Load certificate with password", func(t *testing.T) {
+		provider, err := NewStaticClientCertificateProvider(ClientCertificate{
+			CertFile: "./testdata/test_cert.pem",
+			KeyFile:  "./testdata/test_key_with_thepassword1.pem",
+			Password: &validPassword,
+		})
+		if err != nil {
+			t.Fatalf("Expected no error for valid certificate and key, got %v", err)
+		}
```

Comment:
Could also use `testutil.AssertNoError`. Either way is fine.

Reply:
Cyclical dependency when importing `testutil` package as `testutil` also imports `auth` package for some reason. This needs looked at in another card as it needs refactoring - I don't think `testutil` should depend on auth.